### DISCUSSION TITLE
Server middleware only on Deno Deploy?

### DISCUSSION THREAD

**Discussion by shishkin** [lumeland/lume#315] [Q&A] [answered]
I see that [on_demand](https://lume.land/plugins/on_demand/) only works on Deno Deploy. By looking at the code I see it relies on server and http modules but nothing specific to Deno Deploy. Does it really only work on Deno Deploy and what prevents it from running the server on any other deno server runtime?

**Comment by oscarotero** [comment, marked as answer]
Ok, maybe the message is not correct. It only works on Deno runtimes. In fact, it works better in Deno CLI than Deno Deploy (Deno Deploy has some limitations like not supporting dynamic imports).

Besides Deno Deploy, I don't know any hosting service providing a Deno environment, so this is why I added this warning.

**Reply by shishkin** [reply]
Thanks @oscarotero! That's an encouraging answer 😁 

There are at least options of [AWS Lambda](https://github.com/hayd/deno-lambda), [Vercel](https://github.com/vercel-community/deno) or [Fly](https://fly.io/docs/languages-and-frameworks/deno/) that I'm aware of. And of course people can always run their own Deno process or Docker container on their servers or container schedulers.
